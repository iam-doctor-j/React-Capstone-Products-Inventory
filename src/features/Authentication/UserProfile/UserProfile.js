import React, { useEffect } from 'react';
import Card from "react-bootstrap/Card";
import { connect } from "react-redux";
import { refresh } from "../AuthenticationSlice";

export function UserProfile(props) {
    
    useEffect(() => {
        if(!props.user)
            props.dispatch(refresh());
    }, [])

    return(
        <div className="container">
            <div className="row justify-content-center">
                <Card className="col-12 col-md-8 col-lg-6">
                    <Card.Body>
                        <Card.Title className="mb-4">
                            <div className="d-flex">
                                <span className="heading-text">User Profile</span>
                                <a className="ml-auto" onClick={props.history.goBack} style={{cursor: 'pointer', color: 'var(--secondary)'}}><i className="fas fa-arrow-circle-left"></i></a>
                            </div>
                        </Card.Title>
                        <div className="mb-3">
                            <div className="label-text">First Name</div>
                            <div>{props.user.firstName}</div>
                        </div>
                        <div className="mb-3">
                            <div className="label-text">Last Name</div>
                            <div>{props.user.lastName}</div>
                        </div>
                        <div className="mb-3">
                            <div className="label-text">Email</div>
                            <div>{props.user.email}</div>
                        </div>
                        <div className="mb-3">
                            <div className="label-text">Mobile Number</div>
                            <div>{props.user.mobNumber}</div>
                        </div>
                        <div className="mb-3">
                            <div className="label-text">Location</div>
                            <div>{props.user.location}</div>
                        </div>
                    </Card.Body>
                </Card>
            </div>
        </div>
    );
}

export default connect(state => {
    let user = null;
    if(state.auth.user) {
        user = state.auth.user;
    }

    return { user };
})(UserProfile);