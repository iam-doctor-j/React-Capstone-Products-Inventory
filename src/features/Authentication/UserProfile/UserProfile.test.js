import React from 'react';
import { shallow } from 'enzyme';
import { UserProfile } from './UserProfile';

describe('Test UserProfile with shallow rendering', () => {
    let wrapper;

    const props = {
        dispatch: () => {},
        user: {
            firstName: 'Jyotirmaya',
            lastName: 'Sahu',
            email: 'xyz@xyz.com',
            mobNumber: 1111111111,
            location: 'Bangalore',
        },
        history: {
            goBack: () => {},
        },
    }

    beforeEach(() => {
        wrapper = shallow(<UserProfile {...props}/>);
    });

    it('Should render properly', () => {
        expect(wrapper).toMatchSnapshot();
    })

    it('should render only one container div', () => {
        expect(wrapper.find('div.container').length).toEqual(1);
    })

    it('should display heading text properly', () => {
        expect(wrapper.find('span.heading-text').render().text()).toEqual('User Profile');
    })

    afterEach(() => {
        wrapper.unmount();
    });
})