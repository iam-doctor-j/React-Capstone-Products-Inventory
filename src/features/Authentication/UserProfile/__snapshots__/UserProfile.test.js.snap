// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test UserProfile with shallow rendering Should render properly 1`] = `
<div
  className="container"
>
  <div
    className="row justify-content-center"
  >
    <Card
      body={false}
      className="col-12 col-md-8 col-lg-6"
    >
      <CardBody>
        <CardTitle
          className="mb-4"
        >
          <div
            className="d-flex"
          >
            <span
              className="heading-text"
            >
              User Profile
            </span>
            <a
              className="ml-auto"
              onClick={[Function]}
              style={
                Object {
                  "color": "var(--secondary)",
                  "cursor": "pointer",
                }
              }
            >
              <i
                className="fas fa-arrow-circle-left"
              />
            </a>
          </div>
        </CardTitle>
        <div
          className="mb-3"
        >
          <div
            className="label-text"
          >
            First Name
          </div>
          <div>
            Jyotirmaya
          </div>
        </div>
        <div
          className="mb-3"
        >
          <div
            className="label-text"
          >
            Last Name
          </div>
          <div>
            Sahu
          </div>
        </div>
        <div
          className="mb-3"
        >
          <div
            className="label-text"
          >
            Email
          </div>
          <div>
            xyz@xyz.com
          </div>
        </div>
        <div
          className="mb-3"
        >
          <div
            className="label-text"
          >
            Mobile Number
          </div>
          <div>
            1111111111
          </div>
        </div>
        <div
          className="mb-3"
        >
          <div
            className="label-text"
          >
            Location
          </div>
          <div>
            Bangalore
          </div>
        </div>
      </CardBody>
    </Card>
  </div>
</div>
`;
